dashboard/api.py

```diff
@@ -367,3 +380,127 @@ def format_courserun_for_dashboard(course_run, status_for_user, certificate=None
         # TODO: here goes the logic to pull the current grade  # pylint: disable=fixme
         pass
     return formatted_run
+
+
+def get_student_enrollments(user, edx_client):
```

Comment:
`get_student_enrollments` and `get_student_certificates` are basically the same thing with some minor changes, and the models for `Enrollment` and `Certificate` are identical. we can create a new parent class for those two models (`EdxCachedModel` or something?) and abstract away a ton of the logic for these two `get_` methods. that's a bigger change, of course. given the urgency with this issue, if we want to open up a separate issue to make that change, i'm cool with that. what do you think?